pkg/processor/sumologicschemaprocessor/aggregate_attributes_processor.go

```diff
@@ -0,0 +1,255 @@
+			err := proc.aggregateAttributes(dp.At(i).Attributes())
+			if err != nil {
+				return err
+			}
+		}
+		return nil
+
+	case pmetric.MetricTypeGauge:
+		dp := metric.Gauge().DataPoints()
+		for i := 0; i < dp.Len(); i++ {
+			err := proc.aggregateAttributes(dp.At(i).Attributes())
+			if err != nil {
+				return err
+			}
+		}
+		return nil
+
+	case pmetric.MetricTypeHistogram:
+		dp := metric.Histogram().DataPoints()
+		for i := 0; i < dp.Len(); i++ {
+			err := proc.aggregateAttributes(dp.At(i).Attributes())
+			if err != nil {
+				return err
+			}
+		}
+		return nil
+
+	case pmetric.MetricTypeExponentialHistogram:
+		dp := metric.ExponentialHistogram().DataPoints()
+		for i := 0; i < dp.Len(); i++ {
+			err := proc.aggregateAttributes(dp.At(i).Attributes())
+			if err != nil {
+				return err
+			}
+		}
+		return nil
+
+	case pmetric.MetricTypeSummary:
+		dp := metric.Summary().DataPoints()
+		for i := 0; i < dp.Len(); i++ {
+			err := proc.aggregateAttributes(dp.At(i).Attributes())
+			if err != nil {
+				return err
+			}
+		}
+		return nil
+	}
+
+	return fmt.Errorf("unknown metric type: %s", metric.Type().String())
+}
+
+func (proc *aggregateAttributesProcessor) processTraces(traces ptrace.Traces) error {
+	for i := 0; i < traces.ResourceSpans().Len(); i++ {
+		resourceSpans := traces.ResourceSpans().At(i)
+		err := proc.aggregateAttributes(resourceSpans.Resource().Attributes())
+		if err != nil {
+			return err
+		}
+
+		for j := 0; j < resourceSpans.ScopeSpans().Len(); j++ {
+			scopeSpans := resourceSpans.ScopeSpans().At(j)
+			for k := 0; k < scopeSpans.Spans().Len(); k++ {
+				err := proc.aggregateAttributes(scopeSpans.Spans().At(k).Attributes())
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func (proc *aggregateAttributesProcessor) isEnabled() bool {
+	return len(proc.aggregations) != 0
+}
+
+func (*aggregateAttributesProcessor) ConfigPropertyName() string {
+	return "aggregate_attributes"
+}
+
+func (proc *aggregateAttributesProcessor) aggregateAttributes(attributes pcommon.Map) error {
+	for i := 0; i < len(proc.aggregations); i++ {
+		curr := proc.aggregations[i]
+		names := []string{}
+		attrs := []pcommon.Value{}
+
+		for j := 0; j < len(curr.patternRegexes); j++ {
+			regex := curr.patternRegexes[j]
+			newMap := pcommon.NewMap()
+			newMap.EnsureCapacity(attributes.Len())
+
+			attributes.Range(func(key string, value pcommon.Value) bool {
+				match := regex.FindStringSubmatch(key)
+				if match != nil {
+					// Join all substrings caught by wildcards into one string,
+					// this string will be the name of this key in the new map.
+					// TODO: Potential name conflict to resolve, eg.:
+					// pod_*_bar_* matches pod_foo_bar_baz
+					// pod2_*_bar_* matches pod2_foo_bar_baz
+					// both will be renamed to foo_baz
+					name := strings.Join(match[1:], "_")
+					names = append(names, name)
+					val := pcommon.NewValueEmpty()
+					value.CopyTo(val)
+					attrs = append(attrs, val)
```

Comment:
As mentioned in the comment. This seems to be a general problem with naming, which I'm not sure how to solve.

Reply:
See my comment here: https://github.com/SumoLogic/sumologic-otel-collector/issues/866#issuecomment-1357672421